examples/matt/ctv2/script2.txt

```diff
@@ -0,0 +1,23 @@
+# Exactly two outputs. Since we only care about the final # output key, we
+# don't have to care whether it contains a taproot, has data embedded etc.
+
+# Check output key [i] is [x] tweaked with 
+# [taproot], and no embedded data.
+# [i][d][tapoot][x]
+<key1>
+00 # no taproot to check
+00 # no embedded data
+00 # index 0
+OP_CHECKOUTPUTCONTRACT
+
+# Check output key [i] is [x] tweaked with 
+# [taproot], and no embedded data.
+# [i][d][tapoot][x]
+<key2>
+00 # no taproot to check
+00 # no embedded data
+01 # index 1
+OP_CHECKOUTPUTCONTRACT
+
```

Comment:
Yes. The output amounts are not currently checked, so this example is unsafe allowing someone to add dust amounts to the checked outputs, siphoning off the rest.

There are two ways this could be solved, as mentioned by @bigspider on the [mailinglist](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2023-April/021588.html):

> The most direct and general way to address this would be to allow direct
introspection on the output amounts. This has the complication that output
amounts require 64-bits arithmetics, as discussed in the context of other
proposals, for example: [4].
>
> One more limited approach that works well for many interesting contracts
is that of the deferred checks, implemented in OP_VAULT [2].
The idea is that all the amounts of the inputs that commit to the same
output
script with OP_CHECKOUTPUTCONTRACTVERIFY are added together, and the script
interpreter requires that the amount of that output is not smaller than the
total amount of those inputs.